[visionOS] Only delegate Image Fullscreen to Quick Look if client has proper entitlements

yoelhawa opened the pull request:
#### 6ac0f7beb9713b7e18c58f7cef3a35a57a0b2841
<pre>
[visionOS] Only delegate Image Fullscreen to Quick Look if client has proper entitlements
<a href="https://bugs.webkit.org/show_bug.cgi?id=275235">https://bugs.webkit.org/show_bug.cgi?id=275235</a>
&lt;<a href="https://rdar.apple.com/129366478">rdar://129366478</a>&gt;

Reviewed by Andy Estes.

When going into fullscreen, check that the client has the necessary
entitlements before marking it to be delegated to Quick Look.

* Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenWindowControllerIOS.mm:
(-[WKFullScreenWindowController enterFullScreen:]):
Rename `_isImageElement` to `_isUsingQuickLook` to more accurately
reflect it&apos;s purpose, and perform the entitlement check.

Canonical link: <a href="https://commits.webkit.org/279824@main">https://commits.webkit.org/279824@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/fdb6939b4b6dc428a077eb8555854993a4aa6ffe

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/54646 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/34068 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/7221 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/57926 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/5379 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/41607 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/5387 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/44265 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/3630 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/56741 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/32209 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/47333 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/25390 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/29005 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/3519 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/50749 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/4881 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/59516 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/29884 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/5025 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/51689 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/31029 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/47419 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/51081 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/11977 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/32016 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/30816 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

etiennesegonzac reviewed (commented):
Nice and easy :)
Adding @aestes for the official review!

etiennesegonzac commented on Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenWindowControllerIOS.mm:
```diff
@@ -881,7 +882,7 @@ - (void)enterFullScreen:(CGSize)mediaDimensions
     _parentWindowState = adoptNS([[WKFullScreenParentWindowState alloc] initWithWindow:_lastKnownParentWindow.get()]);
 #if ENABLE(QUICKLOOK_FULLSCREEN)
-    _isImageElement = manager->isImageElement();
+    _isImageElement = manager->isImageElement() && WTF::processHasEntitlement("com.apple.surfboard.chrome-customization"_s);
```
might be a good opportunity to rename `_isImageElement` to `_isUsingQuickLook`

yoelhawa replied:
Good idea!

Commit:
[visionOS] Only delegate Image Fullscreen to Quick Look if client has proper entitlements
https://bugs.webkit.org/show_bug.cgi?id=275235
<rdar://129366478>

Reviewed by Andy Estes.

When going into fullscreen, check that the client has the necessary
entitlements before marking it to be delegated to Quick Look.

* Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenWindowControllerIOS.mm:
(-[WKFullScreenWindowController enterFullScreen:]):
Rename `_isImageElement` to `_isUsingQuickLook` to more accurately
reflect it's purpose, and perform the entitlement check.

Canonical link: https://commits.webkit.org/279824@main
